### PR TITLE
Add Product Discovery section to homepage [MAILPOET-4827]

### PR DESCRIPTION
**Pull request by costasovo**
## Description
This PR adds the product discovery section to the homepage
Screenshot 2023-01-12 at 16 32 19

## Code review notes
I recommend reviewing commit by commit.

## QA notes
When testing using the ZIP build, the illustration images in the task list won't load because they will be publicly available after the release.
You can check the feature with the images by testing with the dev env and adding `define( 'MAILPOET_USE_CDN', false ); ` to the wp-config.php. Alternatively, you can ping me and I'll push the images to the repo so that they also work from the zip.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4827]

## After-merge notes

_N/A_

[MAILPOET-4827]: https://mailpoet.atlassian.net/browse/MAILPOET-4827?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/homepage/components/product-discovery.tsx [hunk 1]
```diff
@@ -87,8 +89,9 @@ export function ProductDiscovery({ onHide }: Props): JSX.Element {
             },
           ]}
         />
-      </div>
-      <ul>{tasks.map((item) => item)}</ul>
-    </div>
+      }
+    >
+      <ul className="mceProgress">{tasks.map((item) => item)}</ul>
```
Comment (lysyjan) [inline mailpoet/assets/js/src/homepage/components/product-discovery.tsx]: The class `mceProgress` looks weird to me. I haven't noticed it in another commit and haven't found it in our codebase.

**Review by lysyjan — COMMENTED** [review]
Great work! I found just one little thing I would like to ask.

File: mailpoet/assets/js/src/homepage/components/product-discovery.tsx (hunk 1 above)
Comment (costasovo) [reply mailpoet/assets/js/src/homepage/components/product-discovery.tsx]: Ha! Nice catch!  I'll remove it. I wonder where it came from. The copilot maybe?

**Comment by costasovo** [comment]
@lysyjan, I rebased the PR on the latest trunk, removed the strange class name, and added one more commit where I slightly improved the homepage section component. I forgot to do it in the original PR.

**Review by lysyjan — APPROVED** [review]

**Comment by Aschepikov** [comment]
Hi @costasovo  there is some error, can you take a look pls
image
`TypeError: Cannot destructure property 'isTaskListHidden' of '(0 , k.Z)(...)' as it is undefined. 
File name: No fileName reported 
Stack trace: TypeError: Cannot destructure property 'isTaskListHidden' of '(0 , k.Z)(...)' as it is undefined.
    at z (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/admin.js?ver=4.4.0:2:8766)
    at ra (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1396148)
    at js (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1448197)
    at xl (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1435471)
    at bl (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1435399)
    at wl (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1435262)
    at fl (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1432249)
    at https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1382070
    at t.unstable_runWithPriority (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1585646)
    at ji (https://brown-parrot.jurassic.ninja/wp-content/plugins/mailpoet/assets/dist/js/commons.js?ver=4.4.0:9:1381847)`

**Comment by costasovo** [comment]
@Aschepikov I think this was caused by cached JS assets.  I changed the naming of some variables ([see commit](https://github.com/mailpoet/mailpoet/pull/4649/commits/3d7e7bd8bcda26a87409b4cfba40834c749583ab)) and if old admin.js file is loaded this is what happens. This caching issue may happen on feature branches because we use the plugin version as a versioning parameter in the asset URL.

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**